### ISSUE TITLE
Can I use mixed image resolution for training?

### ISSUE THREAD

**Issue by itsjustrafli** [ultralytics/yolov5#11643]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am going to training yolov5s model with custom COCO dataset which is 640x480 and I want to add some image dataset (about one eighth of the total dataset) that has resolution of 1280x720.

My question is it possible to do so? If it is possible, is there any setting/argument that has to be set/add before start training? Or it has to be on the same resolution?

Thank you in advance!

**Comment by glenn-jocher** [comment]
@itsjustrafli hello! Yes, you can use mixed image resolutions for training. All you need to do is to specify your dataset paths and image sizes in your training script as arguments. For example, for 640x480 images:

```
python train.py --data coco.yaml --img 640 --batch 16 --epochs 300
```

And for 1280x720 images:

```
python train.py --data coco.yaml --img 1280x720 --batch 16 --epochs 300
```

Note that the batch size should be the same for all resolutions. Let us know if you have any other questions!

**Comment by itsjustrafli** [comment]
Thanks, @glenn-jocher!

Correct me if I'm wrong, but that means the training is done separately? What if I want to do training with the mixed dataset at one time (not separately), is it possible?

Thank you in advance!

**Comment by glenn-jocher** [comment]
Hello, @itsjustrafli! 

If you want to train your model with a mixed dataset of multiple resolutions at once, you can use a method called "rectangular training". This technique involves resizing all the images to one common resolution, usually the smallest resolution in your dataset, and padding the images with zeros to match the aspect ratio of the other, larger images. This way, all images can be resized to the same dimensions, allowing you to train a single model without needing to train on each resolution separately.

To use rectangular training in YOLOv5, you can set the `--rect` argument to True when running the `train.py` script. 

I hope this helps! Let me know if you have any more questions.

**Comment by itsjustrafli** [comment]
Thank you so much @glenn-jocher!

Thanks for the crystal clear explanation!

**Comment by glenn-jocher** [comment]
You're welcome, @itsjustrafli! I'm glad I could provide you with a clear explanation. If you have any more questions or if there's anything else I can help you with, feel free to ask!
